src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorageService.cs

```diff
@@ -66,7 +120,17 @@ protected override string GetDatabaseFilePath(string workingFolderPath)
             }
             catch (Exception)
             {
-                sqlStorage?.Dispose();
```

Comment:
i feel like, at the top of this method, we would see in the first place if we had been able to load sqlite.  And, if not, we wouldn't do *any* work.  no need for locks, exception handling, and wahtnot in that case.  Basically, we shouldn't even be trying this complex logic if we can't even get at sqlite in the first place.

Reply:
This isn't even called if we haven't been able to load sqlite.

Reply:
@Therzok the tests probably are failing because some unrelated tests ran before the Sqlite tests and set the static field to false and Sqlite tests are bailing out due to the static field already being set as false.

I bet before, except Sqlite tests, other conditions made Sqlite service to bail out before creating SqliteStorage (so no static ctor get called) and let the tests to actually run.

...

I think either you take out Sqlite persistent service out from default test MEF so that only SQLite test uses real Sqlite persistent service, or addin test checking code somewhere.

Reply:
or you don't save the flag as you did :) that will work as well.